openwpm/browser_manager.py

```diff
@@ -612,130 +611,158 @@ def shutdown_browser(self, during_init: bool, force: bool = False) -> None:
-        extra = parse_traceback_for_sentry(tb)
-        extra["exception"] = tb[-1]
-        logger.error(
-            "BROWSER %i: Crash in driver, restarting browser manager"
-            % browser_params.browser_id,
-            exc_info=True,
-            extra=extra,
-        )
-        status_queue.put(("FAILED", pickle.dumps(sys.exc_info())))
-    finally:
-        if display is not None:
-            display.stop()
-        return
+                # attempts to perform an action and return an OK signal
+                # if command fails for whatever reason, tell the TaskManager to
+                # kill and restart its worker processes
+                try:
+                    command.execute(
+                        driver,
+                        self.browser_params,
+                        self.manager_params,
+                        extension_socket,
+                    )
+                    self.status_queue.put("OK")
+                except WebDriverException:
+                    # We handle WebDriverExceptions separately here because they
+                    # are quite common, and we often still have a handle to the
+                    # browser, allowing us to run the SHUTDOWN command.
+                    tb = traceback.format_exception(*sys.exc_info())
+                    if "about:neterror" in tb[-1]:
+                        self.status_queue.put(
+                            ("NETERROR", pickle.dumps(sys.exc_info()))
+                        )
+                        continue
+                    extra = parse_traceback_for_sentry(tb)
+                    extra["exception"] = tb[-1]
+                    self.logger.error(
+                        "BROWSER %i: WebDriverException while executing command"
+                        % self.browser_params.browser_id,
+                        exc_info=True,
+                        extra=extra,
+                    )
+                    self.status_queue.put(("FAILED", pickle.dumps(sys.exc_info())))
+
+        except (ProfileLoadError, BrowserConfigError, AssertionError) as e:
+            self.logger.error(
+                "BROWSER %i: %s thrown, informing parent and raising"
+                % (self.browser_params.browser_id, e.__class__.__name__)
+            )
+            self.status_queue.put(("CRITICAL", pickle.dumps(sys.exc_info())))
+        except Exception:
+            tb = traceback.format_exception(*sys.exc_info())
+            extra = parse_traceback_for_sentry(tb)
+            extra["exception"] = tb[-1]
+            self.logger.error(
+                "BROWSER %i: Crash in driver, restarting browser manager"
+                % self.browser_params.browser_id,
+                exc_info=True,
+                extra=extra,
+            )
+            self.status_queue.put(("FAILED", pickle.dumps(sys.exc_info())))
+        finally:
+            if display is not None:
+                display.stop()
```

Comment:
The only difference I see from before is that now that we don't use the [`Process` wrapper](https://github.com/mozilla/OpenWPM/blob/cc10baebee4a0f3fbec38eeac17586448c903e60/openwpm/utilities/multiprocess_utils.py#L35L50) from `utilities.multiprocess_utils` if an exception is raised here it won't be logged. But I don't know if that's really a problem.

Reply:
Yeah, that was my thinking as well. 
I think it's highly unlikely that the display stopping will fail and all other exceptions are caught in the except block above.